flink-core/src/main/java/org/apache/flink/core/fs/AbstractCloseableRegistry.java

```diff
@@ -16,29 +16,29 @@
  * limitations under the License.
  */
 
-package org.apache.flink.runtime.state;
+package org.apache.flink.core.fs;
 
-import org.apache.commons.io.IOUtils;
+import org.apache.flink.util.IOUtils;
 
 import java.io.Closeable;
 import java.io.IOException;
-import java.util.HashSet;
-import java.util.Set;
+import java.util.Map;
 
 /**
- * This class allows to register instances of {@link Closeable}, which are all closed if this registry is closed.
+ * This is the abstract base class for registries that allow to register instances of {@link Closeable}, which are all
+ * closed if this registry is closed.
  * <p>
  * Registering to an already closed registry will throw an exception and close the provided {@link Closeable}
  * <p>
  * All methods in this class are thread-safe.
  */
-public class ClosableRegistry implements Closeable {
+public abstract class AbstractCloseableRegistry<C extends Closeable, T> implements Closeable {
```

Comment:
Could you please document the two generic arguments. I now know what they are for but I think future readers of this will find it easier if they are documented.